lib/plugins/aws/package/compile/events/stream/index.js

```diff
@@ -11,40 +11,57 @@ class AwsCompileStreamEvents {
       'package:compileEvents': this.compileStreamEvents.bind(this),
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8034
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'stream', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      anyOf: [
+        { $ref: '#/definitions/awsArnString' },
+        {
+          type: 'object',
+          properties: {
+            arn: { $ref: '#/definitions/awsArn' },
+            type: { enum: ['dynamodb', 'kinesis'] },
+            batchSize: { type: 'integer', minimum: 1, maximum: 10000 },
+            parallelizationFactor: { type: 'integer', minimum: 1, maximum: 10 },
+            startingPosition: { enum: ['LATEST', 'TRIM_HORIZON'] },
+            enabled: { type: 'boolean' },
+            consumer: { oneOf: [{ const: true }, { $ref: '#/definitions/awsArn' }] },
+            batchWindow: { type: 'integer', minimum: 0, maximum: 300 },
+            maximumRetryAttempts: { type: 'integer', minimum: -1, maximum: 10000 },
+            bisectBatchOnFunctionError: { type: 'boolean' },
+            maximumRecordAgeInSeconds: {
+              oneOf: [
+                { type: 'integer', minimum: -1, maximum: -1 },
+                { type: 'integer', minimum: 60, maximum: 604800 },
+              ],
+            },
+            destinations: {
+              type: 'object',
+              properties: {
+                onFailure: {
+                  oneOf: [
+                    { $ref: '#/definitions/awsArnString' },
+                    {
+                      type: 'object',
+                      properties: {
+                        arn: { $ref: '#/definitions/awsArn' },
+                        type: { enum: ['sns', 'sqs'] },
+                      },
+                      additionalProperties: false,
+                      required: ['arn'],
```

Comment:
I think we should also require `'type'`

Reply:
type is only required when arn is not provided as a string, otherwise it can be computed from the string, thus making it conditionally required. However, I did not managed to achieve such conditional dependency in JSON schema. The properties dependency feature does not allow the same property to have multiple definition.

Reply:
I used `anyOf` to prevent such issue.

I updated with 3 possible cases. The same schema was also implemented for `stream.arn` and `stream.type` property at root. In order to avoid too much duplication, I added a `addDefinitionsToSchema` function on `configSchemaHandler` and defined mutual schema specific to stream event in this function. WDYT ?